aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,398 @@
+        conf.getBoolean(IcebergAWSConfigs.ICEBERG_AWS_GLUE_SKIP_ARCHIVE,
+            IcebergAWSConfigs.ICEBERG_AWS_GLUE_SKIP_ARCHIVE_DEFAULT)
+    );
+  }
+
+  /**
+   * Constructor with default glue client
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   * @param skipArchive if Glue should skip an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to turn this feature off.
+   */
+  public GlueCatalog(
+      Configuration conf,
+      String catalogId,
+      String warehousePath,
+      boolean skipArchive) {
+    this(conf, GlueClient.create(), catalogId, warehousePath, skipArchive);
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  /**
+   * All-arg constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   * @param skipArchive if Glue should skip an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to turn this feature off.
+   */
+  public GlueCatalog(
+      Configuration conf,
+      GlueClient glue,
+      String catalogId,
+      String warehousePath,
+      boolean skipArchive) {
+    this.conf = Preconditions.checkNotNull(conf, "Cannot initialize GlueCatalog because Hadoop config is null");
+    this.glue = Preconditions.checkNotNull(glue, "Cannot initialize GlueCatalog because GlueClient is null");
+    this.catalogId = catalogId;
+    this.catalogName = "glue." + catalogId; // for logging purpose
+    this.warehousePath = warehousePath;
+    this.skipArchive = skipArchive;
+    validateWarehousePath();
+  }
+
+  private void validateWarehousePath() {
+    Preconditions.checkArgument(warehousePath != null && warehousePath.length() > 0,
+        "Cannot initialize GlueCatalog because warehousePath must not be null");
+    Preconditions.checkArgument(warehousePath.charAt(warehousePath.length() - 1) != '/',
+        "Cannot initialize GlueCatalog because warehousePath %s must not end with /", warehousePath);
```

Comment:
Why not just remove the trailing `/`?

Reply:
yeah that's actually better, let me do that